### PR TITLE
#167728018 User can login with email and password

### PR DESCRIPTION
**Pull request by justinefe**
**What does this PR do?**
 Creates user can login

  **Description of Task to be completed?**
 >- Set up unit test
  >- Perform validation
  >- Creates a token on sign in
  >- Decrypts user's password
  >- Created user's sign in bussines controller
  >- POST /api/v1/auth/signin
  
  **How should this be manually tested?**
 >- Open command terminal 
 >- Clone this repository
 >- Run npm install 
 >- Run npm test
>- Start server
>- Open postman

**What are the relevant stories?**
#167728018

 **Questions:**
none

### CODE REVIEW

File: src/middlewares/userAuth.js [hunk 1]
```diff
@@ -34,4 +34,28 @@ export default class userAuth {
...
+
+    const schema = {
+      email: inValidEmail(email),
+      password: inValidPassword(password),
+    };
+    const error = validate(schema);
+    if (error) return sendErrorResponse(res, 422, error);
+    return next();
+
+  }
```
[the remark this answers is not in the document]
Comment (Kingobaino) [reply src/middlewares/userAuth.js:58]: Good one... I think you also need to check if the account is verified too..

**Review by Kingobaino — COMMENTED** [review]
I think that you need to check weather the account is verified before a user can sign in..

File: src/helpers/token.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const secretKey = process.env.JWT_SECRET;
+const token = payload => jwt.sign(payload, secretKey, {
+  expiresIn: '1d',
```
Comment (dieudonneAwa) [inline src/helpers/token.js]: I don't think you need this file again. This has been done and merged

File: src/controllers/UserController.js [hunk 3]
```diff
@@ -13,6 +19,40 @@ class UserController {
...
+        if (checkUser.is_verified === false) return sendErrorResponse(res, 401, 'Authentication Error');
+        const isVerified = unhash(password, checkUser.password);
+        if (isVerified) {
+          return res.status(200).json({
+            status: 200,
+            data: {
+              token: token({ id: checkUser.id }),
+              id: checkUser.id,
+              first_name: checkUser.first_name,
+              last_name: checkUser.last_name,
```
Comment (dieudonneAwa) [inline src/controllers/UserController.js]: I think your table fields are not complete. And we agreed not to use `first_name `and `last_name `again. We agreed to use just `name`

File: src/repositories/UserRepository.js [hunk 4]
```diff
@@ -94,4 +96,25 @@ export default class UserRepository {
     });
     return userInfo(dataValues);
   }
+
+  static async find({ email, password }) {
+    const checkUser = await User.findOne({ where: { email } });
+
+    if (!checkUser) throw new Error('User not found');
```
Comment (Abobos) [inline src/repositories/UserRepository.js]: These logics should be in your controller.. I think the repository is just to handle database calls

File: src/controllers/UserController.js (hunk 3 above)
Comment (Wokoro) [reply src/controllers/UserController.js]: Ya. That's true.

**Review by Wokoro — COMMENTED** [review]
I think  when hound errors are fixed it looks good.

**Review by Wokoro — APPROVED** [review]
LGTM

**Review by Abobos — APPROVED** [review]

**Review by Wokoro — APPROVED** [review]
LGTM

**Review by dieudonneAwa — APPROVED** [review]

**Review by chukwuemekachm — APPROVED** [review]

**Pull request merged**